src/bin/cargo-plumbing/plumbing/locate_project.rs

```diff
@@ -0,0 +1,40 @@
+use std::env;
+use std::fmt::Debug;
+use std::path::PathBuf;
+
+use cargo::util::important_paths::find_root_manifest_for_wd;
+use cargo::GlobalContext;
+use cargo_plumbing::CargoResult;
+use serde::Serialize;
+
+#[derive(Debug, clap::Args)]
+pub(crate) struct Args {
+    /// Path to the working directory
+    #[arg(long)]
+    working_directory: Option<PathBuf>,
```

Comment:
Cargo has *unstable* support for this as `-C` but it also does a `std::set_current_dir` before anything else is executed.

We should probably hold off on innovating unless we have a driving need for it which should be clearly communicated in the PR description.